src/python/utils/dotdict.py

```diff
@@ -0,0 +1,18 @@
+# encoding: utf-8
+# author:   Jan Hybs
+
+class DotDict(dict):
+    def __init__(self, *a, **kw):
+        dict.__init__(self, *a, **kw)
+
+    def __getattribute__(self, name):
+        try:
+            return dict.__getattribute__(self, name)
+        except AttributeError:
+            if name in self:
+                return self[name]
+
+            name = name.replace('_', '-')
```

Comment:
I don't know the exact usage point of this class, but this substitution seems to be confusing in general usage since it relax strong consistency in variable/key names.